### ISSUE TITLE
Target specific list values

### ISSUE THREAD

**Issue by BuggyAl** [SkriptLang/Skript#5030]
### Suggestion

I think it would be useful to be able to target specific values in a list.

If we store a location in each world, it may be stored under `{players::UUID::WORLD::location}`. What if I wanted to get a list of all player locations in the end dimension? 
```
Current Skript
loop {players::*}:
    add {players::%loop-index%::end::location} to {_locationlist::*}
```
```
New Syntax
set {_locationlist::*} to {players::*::end::location}
```

Another example is a quest system. What if I wanted the list of all items required for every quest?
```
Current Skript
loop {quest::*}:
    add {quest::%loop-index%::*} to {_questItems::*}
```
```
New Syntax
set {_questItems:*} to {quest::*::items::*}
```

### Why?

I want this feature added because it would make list management much easier and allows developers to clean up their redundant loops.

### Agreement

- [X] I have read the guidelines above and affirm I am following them with this suggestion.

**Comment by TheLimeGlass** [comment]
Yes this is a good suggestion, but would make pattern handling very complex. What if you had multiple locations, but also wanted multiple players `{players::*::locations::*}` at that point you wouldn't have a reference to the player, it'll just be a list of locations and you wouldn't know where it came from. Maybe only allow one asterisk in the pattern to avoid this.

This can be avoided with `indices of {_list::*}` and looping the indices with another list call.
```
loop indicies of {player::*}:
    set {_locations::*} to {player::%loop-value%::locations::*}
```
